models/doctype/Email/EmailForm.js

```diff
@@ -0,0 +1,22 @@
+const BaseForm = require('frappejs/client/view/Form');
+const frappe = require('frappejs');
+
+module.exports = class EmailForm extends BaseForm {
+    constructor({ doctype, parent, submit_label='', container, meta, inline=false }) {
+        super({ doctype: 'Email', parent: parent, container: container, meta: meta });
+        this.makeToolbar();
+    }
+
+    makeToolbar() {
+        this.sendButton = this.container.addButton(frappe._("Send"), 'primary', async (event) => {
+           frappe.call({
+               method: 'send-mail',
+               args: this.doc.getValidDict()
+           });
+        });
+        this.on('change', () => {
+            const show = this.doc._dirty && !this.doc.submitted;
```

Comment:
Recheck if this on change listener is correctly implemented or needed (?)

Reply:
Gotcha 🙈  . Resolved! . now we aren't explicitly doing an `insert` . 
`await this.save` does the job .

Reply:
@surajshetty3416 only doubt . `this.doc.name` in this case will be some random string . i want to assign it to the return value of the function call  (`frappe.call` thing) . i tried this doesn't work 🙈